src/app/screens/Onboard/TestConnection/index.tsx

```diff
@@ -13,36 +13,44 @@ export default function TestConnection() {
     name: string;
     balance: number;
   }>();
-  const [errorMessage, setErrorMessage] = useState();
+  const [errorMessage, setErrorMessage] = useState("");
   const [loading, setLoading] = useState(false);
   const { t } = useTranslation(["welcome"]);
 
   const navigate = useNavigate();
 
-  function handleEdit(event: React.MouseEvent<HTMLButtonElement>) {
-    utils.call("removeAccount").then(() => {
-      navigate(-1);
-    });
+  async function handleEdit(event: React.MouseEvent<HTMLButtonElement>) {
+    await utils.call("deleteAccount");
+    navigate(-1);
   }
 
-  function loadAccountInfo() {
+  async function loadAccountInfo() {
     setLoading(true);
-    api
-      .getAccountInfo()
-      .then((response) => {
-        const name = response.name;
-        const { alias } = response.info;
-        const { balance: resBalance } = response.balance;
-        const balance =
-          typeof resBalance === "number" ? resBalance : parseInt(resBalance);
+    try {
+      const response = await api.getAccountInfo();
 
-        setAccountInfo({ alias, balance, name });
-      })
-      .catch((e) => {
+      if ("error" in response) {
```

Comment:
when is this the case? doesn't getAccountInfo throw an error in the error case?

Reply:
Yeah, thanks. Had a look again and changed it.